### DISCUSSION TITLE
Support StdErr for interactive inputs

### DISCUSSION THREAD

**Discussion by lammel** [pterm/pterm#522] [Ideas]
When writing CLI applications (at least under Linux/Mac) using the pipe to either feed or read from/to StdErr is a quite common operation.

Currently this is quite hard with PTerms interactive input, as currently on area is used for input, which is not prepared for  specifying a writer or support StdErr for reading/writing.

A simple example would be a CLI piping it's own output to stdin, which requires an interactive input (like reading a password) to continue:

```
sh# mycli export something | mycli import something
```

Are there any plans to support StdErr, should I open a draft PR to work on that issue?

**Comment by MarvinJWendt** [comment]
Sure, you can make a proposal in the issues. Now that cursor has better support for changing the output target, this shouldn't be a problem :)

**Reply by lammel** [reply]
As we are working on https://github.com/atomicgo/cursor/pull/17#issuecomment-1599138517 in the underlying cursor library, I'll open a new PR for pterm once this is merged.
Some early testing showed good results with some minor tweaks required for multiline inputs and making windows and linux both work.
